target/linux/ramips/dts/mt7621_buffalo_wsr-2533dhpl.dts

```diff
@@ -0,0 +1,207 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "buffalo,wsr-2533dhpl", "mediatek,mt7621-soc";
+	model = "Buffalo WSR-2533DHPL";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_diag;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+		label-mac-device = &ethernet;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,57600";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		internet_green {
+			label = "wsr-2533dhpl:green:internet";
+			gpios = <&gpio1 9 GPIO_ACTIVE_LOW>;
+		};
+
+		router_green {
+			label = "wsr-2533dhpl:green:router";
+			gpios = <&gpio1 10 GPIO_ACTIVE_LOW>;
+		};
+
+		router_amber {
+			label = "wsr-2533dhpl:amber:router";
+			gpios = <&gpio1 11 GPIO_ACTIVE_LOW>;
+		};
+
+		internet_amber {
+			label = "wsr-2533dhpl:amber:internet";
+			gpios = <&gpio1 12 GPIO_ACTIVE_LOW>;
+		};
+
+		wireless_amber {
+			label = "wsr-2533dhpl:amber:wireless";
+			gpios = <&gpio1 13 GPIO_ACTIVE_LOW>;
+		};
+
+		led_power: power {
+			label = "wsr-2533dhpl:green:power";
+			gpios = <&gpio1 14 GPIO_ACTIVE_LOW>;
+		};
+
+		led_diag: diag {
+			label = "wsr-2533dhpl:amber:diag";
+			gpios = <&gpio1 15 GPIO_ACTIVE_LOW>;
+		};
+
+		wireless_green {
+			label = "wsr-2533dhpl:green:wireless";
+			gpios = <&gpio1 16 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 3 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		aoss {
+			label = "aoss";
+			gpios = <&gpio0 6 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+
+		auto {
+			label = "auto";
+			gpios = <&gpio0 7 GPIO_ACTIVE_LOW>;
+			linux,code = <BTN_0>;
+			linux,input-type = <EV_SW>;
+		};
+
+		bridge {
+			label = "wb";
+			gpios = <&gpio0 10 GPIO_ACTIVE_LOW>;
+			linux,code = <BTN_1>;
+			linux,input-type = <EV_SW>;
+		};
+
+		router {
+			label = "router";
+			gpios = <&gpio0 12 GPIO_ACTIVE_LOW>;
+			linux,code = <BTN_1>;
```

Comment:
BTN_2? Or are these two positions of the same switch? (which we most probably cannot implement)

Reply:
- slide-switch1 (2x positions)
  - "AUTO"
  - "MANUAL" (not connected to gpio)

- slide-switch2 (3x positions)
  - "ROUTER"
  - "AP" (not connected to gpio)
  - "WB"